src/main/scala/io/github/spark_redshift_community/spark/redshift/Conversions.scala

```diff
@@ -30,6 +32,13 @@ import org.apache.spark.sql.types._
  */
 private[redshift] object Conversions {
 
+  /**
+    * From the DateTimeFormatter docs (Java 8):
+    * "A formatter created from a pattern can be used as many times as necessary,
+    * it is immutable and is thread-safe."
+    */
+  private val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss[.SSS][.SS][.S][X]")
```

Comment:
There could be more than 3 `S` here, it's causing errors in cases like `2018-05-03 06:23:33.081834+00`

Reply:
The backend uses [Calendar](https://docs.oracle.com/javase/8/docs/api/java/util/Calendar.html#MILLISECOND), which won't go beyond milliseconds. Most of spark-redshift code is meant (and tested) not to go beyond milliseconds: https://github.com/spark-redshift-community/spark-redshift/blob/master/src/test/scala/io/github/spark_redshift_community/spark/redshift/ConversionsSuite.scala#L48

Reply:
But it did in my case. I had to change the fetched format in my case as a quick fix.

Reply:
You mean, you changed the format of your columns ?
Let me know if you'd like to take a crack at this to fix #35

Reply:
Yes I had the change the format of my columns.
I can help with a fix, but I'm really not familiar with best practices here, if adding more `S` here fixes it then fine.

Reply:
That's not enough. The backend needs to upgrade to something else than the [Calendar](https://docs.oracle.com/javase/8/docs/api/java/util/Calendar.html#MILLISECOND) library or we'll stay limited to milliseconds. So it requires a backend library change plus updating the string format.

Reply:
It was working fine with databricks spark-redshift 3.0.0-preview

Reply:
maybe we can truncate to 3 digits, and show a warning (notice) about that and avoid an exception